ros2doctor/ros2doctor/api/package.py

```diff
@@ -0,0 +1,112 @@
+    url = rosdistro.get_index_url()
+    i = rosdistro.get_index(url)
+    distro_data = rosdistro.get_distribution(i, distro_name).get_data()
+    return distro_data.get('repositories')
+
+
+class PackageCheck(DoctorCheck):
+    """Check local package versions against release versions on rosdistro."""
+
+    def category(self):
+        return 'package'
+
+    def check(self):
+        """Check packages within the directory where command is called."""
+        result = Result()
+        try:
+            distro_packages_info = get_ros_packages_info()
+        except AttributeError:
+            result.add_error('ERROR: Unable to obtain current distro package information \
+                from rosdistro.')
+            return result
+        local_packages_prefixes = get_packages_with_prefixes()
+        if not local_packages_prefixes or not distro_packages_info:
+            result.add_error('ERROR: local packages info or distro packages info is empty.')
+            return result
+        for package_name, package_prefix in local_packages_prefixes.items():
+            file_path = os.path.join(package_prefix, 'share', package_name)
+            if package_name in distro_packages_info:
+                try:
+                    package_info = distro_packages_info.get(package_name)
+                    required_ver = package_info.get('release').get('version')
+                except AttributeError:
+                    result.add_warning('`%s` is not found in distribution release list.'
+                                       % package_name)
+                    required_ver = ''
+            try:
+                package_obj = parse_package(file_path)
+                local_ver = package_obj.version
+            except (AttributeError, IOError, InvalidPackage):
+                result.add_warning('Unable to parse `%s` package.xml file.' % package_name)
+                local_ver = ''
+            if required_ver and local_ver:
+                if required_ver[:3] != package_obj.version[:3]:
+                    result.add_warning('%s local version %s does not match required version %s.'
+                                       % (package_name, package_obj.version, required_ver))
+        return result
+
+
+class PackageReport(DoctorReport):
+    """Report local package versions and release versions on rosdistro."""
+
+    def category(self):
+        return 'package'
+
+    def report(self):
+        """Report packages within the directory where command is called."""
+        report = Report('PACKAGE VERSIONS')
+        try:
+            distro_packages_info = get_ros_packages_info()
+        except AttributeError:
+            return report
+        local_package_prefixes = get_packages_with_prefixes()
+        if distro_packages_info and local_package_prefixes:
+            for package_name, package_prefix in local_package_prefixes.items():
+                try:
+                    package_info = distro_packages_info.get(package_name)
```

Comment:
Same comment about `distro_packages_info` being repository info, and this not working for repos with multiple packages.